### ISSUE TITLE
Will you make a version for Foundation 6?

### ISSUE THREAD

**Issue by klausbrinch** [olefredrik/FoundationPress#556]
Foundation 6 just launched today, so i was wondering whether you have any plans of supporting Foundation 6, and if so when. And thank you for FoundationPress

**Comment by olefredrik** [comment]
FoundationPress will be updated to the Foundation 6 soon. Older versions (based on Foundation 5) will still be available in the release archive when the new version is released.

**Comment by emiliog07** [comment]
Foundation 6 looks incredible! Can't wait to see your Foundation6Press :+1:

**Comment by olefredrik** [comment]
I agree there. Great kudos to the team at Zurb. They've done a massive job with the F6 release. I like what I've been seing so far. That said, I have found that it is always wise to wait a little while before starting migration to new major versions, as there usually are many updates and patches the first few weeks after a major release.

**Comment by jonasopdebeeck** [comment]
Thumbs up for making the effort of upgrading, @olefredrik ! Looking forward to FoundationPress based on Foundation 6!

**Comment by LukePettway** [comment]
@olefredrik thanks again for your contributions, I continue to use FoundationPress on every WP site I build and the addition of Foundation 6 is going to make everything even better! :dancers:

**Comment by olefredrik** [comment]
:+1:

**Comment by illusivesunrae** [comment]
@olefredrik Do you have a rough estimate on when you'll have an updated version? Thanks! :)

**Comment by AdamChlan** [comment]
@olefredrik I don't know your plans to roll out the theme, but if it makes sense to set up task/issues in the tracker, I'd be happy to contribute to get tasks completed.

**Comment by t-fulton** [comment]
Smart about waiting to the kinks are ironed out, thanks for a fantastic theme!

**Comment by colin-marshall** [comment]
@olefredrik if you need assistance with the migration or beta testers for the Foundation 6 version of FoundationPress, please let us know how we can help.

**Comment by olefredrik** [comment]
Thanks @colin-marshall ! I'm in the process of upgrading FP to Foundation v6.0.1 as we speak. So far it looks good. But there is some work to be done before it is ready for release. I will push the development branch when it's ready for beta testing. Stay tuned :)

**Comment by colin-marshall** [comment]
Thanks!

**Comment by tomhermans** [comment]
Wanted to know this as well. Thanks for the reply I was hoping for. Awesome work !

**Comment by olefredrik** [comment]
Current status: Attempting to be smarter than I actually am. Build script now compiles all F6 assets without errors. Next: Topbar and offcanvas navigation. And coffee. Lots of coffee.

![2015-11-21 00 58 55](https://cloud.githubusercontent.com/assets/1080017/11314957/372db7b8-8feb-11e5-94a6-7815db6d8b72.jpg)

**Comment by olefredrik** [comment]
[Current status](https://pbs.twimg.com/tweet_video/CUSVoGIWUAAX2yH.mp4)  <-- Not too far away from the truth, actually.

But seriously. I have made some progress through the weekend. There are still some challenges around top bar and offcanvas navigation. It works, but far from perfect.  I also have some challenges with Magellan and Sticky plugins. 

I don't want to drop a new major release until everything has been tested and works fairly well.

**Comment by arielnoname** [comment]
@olefredrik Nice work man, and very nice working place. Thanks for taking the time so we all can enjoy this.

**Comment by Aetles** [comment]
Well, Zurb have already released four versions of Foundation 6 in just a couple of days (6.0.0 -> 6.0.3) so it is probably a good thing to take your time and not rush out a new major release.

**Comment by LukePettway** [comment]
@olefredrik , let me buy you a coffee!

**Comment by stirlingsohn** [comment]
yeah where can we donate? FoundationPress is King!

**Comment by olefredrik** [comment]
Haha, thank you! The whole project is based on good teamwork, so it would be wrong to accept donations. But if you are happy with FP, please feel free to donate a buck for a good cause (Red Cross, Amnesty International, or similar) on behalf of yourself and all awesome contributors to the project.

**Comment by dantahoua** [comment]
Thanks a lot for your work! FoundationPress saved me some precious time. With Foundation 6 it seems more time will be saved (with animation and menu!).

**Comment by jimlongo56** [comment]
Really looking forward to FoundationPress with Foundation 6. 

We're using a slightly customized version on all our sites now - adding required plugins and added dependencies. We're even requiring outside design shops to start with a branch of our FoundationPress. Makes maintenance and further development a snap. 

Thanks for your efforts on this!

**Comment by josh-rathke** [comment]
Thanks for your work on this @olefredrik, this starter theme is one for the books and bringing into the 6.0 generation is only going to make it stronger!

**Comment by drivenfaroff** [comment]
Thanks for everything you do. Looking forward to the version 6 release!

**Comment by Eccola** [comment]
Thank you so much for preparing these goodies for us @olefredrik. 
What puzzles me, is FoundationPress going to remain grunt-powered in the future?

**Comment by olefredrik** [comment]
@Eccola : As of now, I'm still using Grunt. Although Zurb have switched to Gulp on Foundation 6, I'm not sure if we were going to replace something that works for us. Both Grunt and Gulp works quite well, imo. As long as we the libsass, compiling will be fast either we use one or the other. My focus now is not on switching task runner technology, but to get all F6 components to work with WordPress. But I'm open to a discussion around this, if anyone has really good arguments for switching to Gulp.

**Comment by alexbohariuc** [comment]
@olefredrik count me in for testing.
Great job!

**Comment by radel** [comment]
@olefredrik the only argument I can think of for switching to Gulp is consistency with foundation, both in workflow and directory tree. We could take the good work the foundation team has done and use it in FoundationPress with little adjustments. In the long run, keeping up with Grunt could be much harder than just pull the gulpfile from foundation-sites. 
Also, I can't wait to get my hands dirty and help you in a development branch :)

keep up the great work!

**Comment by colin-marshall** [comment]
I vote for a switch to Gulp. The consistency argument is an important one. A fair number of your users probably learned and used Foundation before learning and using FoundationPress. Those people will be more comfortable with FoundationPress from the get go if you're using the same build system that they learned to use with Foundation. A lot of users will never touch the build system, but you'll make life easier for those who do if you are on the same task runner/build system that Foundation is using.

**Comment by Eccola** [comment]
I'd love to have "the original" FoundationPress kept intact from the task runner change.
How's about creating a new branch of FoundationPress with gulp build bells and whistles?
It would sure grow the user and the contributor base.

**Comment by radel** [comment]
@Eccola they can coexist, in the end they are just two files in your project root and and you can run whatever you want.

**Comment by jimlongo56** [comment]
I'm familiar with the current process, and lazy, so stay with Grunt.  No big deal either way. 

As far as I can see, the main pro for Gulp is speed, but the time taken for the tasks I have running currently is not slowing my workflow down.

**Comment by josh-rathke** [comment]
The original FoundationPress would always be available through the last
release using Grunt if the change to Gulp was ever made. I think trying to
manage two different Forks would be pretty inefficient. One would end up
lagging WAY behind. I would say if it's time to move on, then it's time to
move on.

On Friday, November 27, 2015, Eccola <notifications@github.com> wrote:

> I'd love to have "the original" FoundationPress kept intact from the task
> runner change.
> How's about creating a new branch of FoundationPress with gulp build bells
> and whistles?
> It would sure grow the user and the contributor base.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/olefredrik/FoundationPress/issues/556#issuecomment-160147582>
> .
>


-- 
Josh Rathke
I.T. Director & Web Developer, YWAM Montana | Lakeside
p: (406) 844-2221 m: (765) 637-5847 a: 501 Blacktail Rd. Lakeside MT, 59922
s: www.ywammontana.org e: josh.rathke@ywammontana.org

**Comment by colin-marshall** [comment]
2 forks would be a pain, I agree.

Maybe the question if Grunt should be replaced by Gulp in FoundationPress should also be asked to the Foundation devs at Zurb. They would know better than anybody else if there would be an issue/conflict down the road with FoundationPress continuing to use Grunt for future releases.

**Comment by josh-rathke** [comment]
I think there is a conversation on whether to move to Gulp or not. But I think that's where it needs to end.

I agree with Colin. Multiple Forks would present some issues in terms of slowing down feature releases due to having to make everything work on both. The alternative is one quietly dies anyway, while a bunch of people get frustrated it isn't ever being updated.

I also wouldn't want to have multiple task runners within the project side by side. That get's confusing documentation-wise, and could potentially screw peoples projects up if they work on many different projects and switch from one to another, then have to remember which one they were using in a "FoundationPress Built" project.

Like I said above, my vote would be that if we decide consistency with Zurb is king, then let's not try to hang on to the past. Let's move on full steam ahead. Old projects will still function just fine. New projects will function better.

**Comment by arielnoname** [comment]
I think the consistency argument takes it all. I've never used Gulp before but if Zurb is changing Grunt and to that we sum up the speed thing, it's a winner.

**Comment by ryankidd** [comment]
Looking forward to it.  Thanks team FP!

**Comment by jorcons** [comment]
I prefer Grunt but if you decide move to Gulp it's ok. Great work guys! I use FoundationPress on most of my projects. It's awesome!

**Comment by Tralapo** [comment]
Don't have much to say about Grunt vs. Gulp, I'm fine with both I think. Keep up the good working, looking forward to FP based on F6!

**Comment by Aetles** [comment]
For anyone familiar with Grunt but not Gulp, here's a good start: [Switching from Grunt to Gulp](http://www.techinsight.io/review/devops-and-automation/switching-from-grunt-to-gulp/).

**Comment by swthate** [comment]
Thanks for the link, @Aetles!

*saves to favorites*

**Comment by jorcons** [comment]
Good article @Aetles. Thanks! I'm looking forward too for FP with F6.

**Comment by mhair** [comment]
Looking forward to the beta testing the development branch as well! Great starter theme, and Foundation 6 is  going to be a nice step forward!

**Comment by imaginethepoet** [comment]
Like the rest here happy to test or look at some code tasks. I've been working with foundation 6 a little before public release. Share the challenges you are having and i believe this group will help out.  If you have a beta branch up we can look at that.  Great work.

**Comment by olefredrik** [comment]
Ta-da!  https://github.com/olefredrik/FoundationPress/pull/580

**Comment by jorcons** [comment]
Esta dirección en breve dejará de funcionar.
Rogamos tome nota del nuevo email: jconstanti@visibletic.com

Visible IT Comunicación
visibletic.com

**Comment by swthate** [comment]
And there was much rejoicing!!

Bravo

![Much Rejoicing](http://cdn.makeagif.com/media/4-19-2015/9R3g9S.gif)

**Comment by olefredrik** [comment]
I have not prioritized a switch from Grunt to Gulp yet. But it may be added over time. Enjoy! :)

**Comment by josh-rathke** [comment]
This is awesome! Thank you for what had to be hours and hours of tinkering!

Josh Rathke
I.T. Director & Web Developer, YWAM Montana | Lakeside
p: (406) 844-2221 m: (765) 637-5847 a: 501 Blacktail Rd. Lakeside MT, 59922
s: www.ywammontana.org e: josh.rathke@ywammontana.org

**Comment by imaginethepoet** [comment]
Great

**Comment by colin-marshall** [comment]
@olefredrik: Nice work! I've already started adding Gulp on a branch of my [FoundationPress Fork](https://github.com/colin-marshall/FoundationPress/tree/f6-gulp). I have the JS and Sass tasks working and will work on the rest over the weekend. I'll make a PR when it's more complete so you can check it out and decide if you want to use it.

@josh-rathke: Hello fellow Montanan! Missoulian here. Never in a million years did I think I would see anybody else from Montana on this. Cheers!!!

**Comment by olefredrik** [comment]
@colin-marshall : Sweet! Looking forward to reviewing your pull request with Gulp :+1:

**Comment by colin-marshall** [comment]
Just thought I would post one last time in this thread to let everybody know that the switch to Gulp has happened. Test it out and feel free to @ me in any Gulp related issues you open so I can assist.

**Comment by jorcons** [comment]
Great Collin! Thanks for you work. I'll try and give you feedback.

**Comment by josh-rathke** [comment]
@colin-marshall That's awesome! And yeah that's crazy about you being in Montana as well. Definitely not a lot of coders out here. I have met a couple pretty darn good ones up here in Kalispell though.

Nice work man, this will make future projects incredible!

**Comment by Aetles** [comment]
Just a quick follow-up, looking back at this now…

Back in November when we discussed the move to Foundation 6 I was [a little cautious](https://github.com/olefredrik/FoundationPress/issues/556#issuecomment-158817238) not to rush out this new major release since I felt that Zurb was still moving a little too fast. Looking back, I think Foundation 6 has hade somewhat of a rocky road since then, compared to the stability we were used to with Foundation 5. This has now been acknowledged by Zurb in [an answer](https://github.com/zurb/foundation-sites/issues/8324#issuecomment-197072342) to some frustrated users in another issue:

> We hear your issues—know that we're working on it. We moved a little too fast the last three months, and our intent for the near future is to slow down and stabilize.

This is good news for us. They have also started with more testing and intend to bring in more people from the community:

> You might have noticed that we've started writing visual regression tests as we address bugs. We've also published a full testing guide and started an initiative to build out a JavaScript test suite.

> We also intend to soon bring in members of the community to be direct contributors to the codebase, help review pull requests, and so on. So we're gradually moving towards a system where decisions are made in a more deliberate manner, by a larger group of people. Foundation 6 was an aggressive reset to the structure of the codebase, with an eye towards making it more sustainable long-term.

**Comment by jorcons** [comment]
Esta dirección en breve dejará de funcionar.
Rogamos tome nota del nuevo email: jconstanti@visibletic.com

Visible IT Comunicación
visibletic.com
